src/instructlab/model/backends/vllm.py

```diff
@@ -144,3 +133,34 @@ def run_vllm(
     logger.info(f"vLLM starting up on pid {vllm_process.pid} at {api_base}")
 
     return vllm_process
+
+
+def build_vllm_cmd(
+    host: str, port: int, model_path: pathlib.Path, vllm_args: list[str]
+) -> list[str]:
+    """
+    Build the vLLM command to run the server.
+    """
+    vllm_cmd = [sys.executable, "-m", "vllm.entrypoints.openai.api_server"]
+
+    # The following line handles the case where the arg is passed like ["--option=value"]
+    # where typically we expect it to be ["--option", "value"].
+    arg_names = set(e.split("=", 1)[0] for e in vllm_args)
+    if "--host" not in arg_names:
+        vllm_cmd.extend(["--host", host])
+
+    if "--port" not in arg_names:
+        vllm_cmd.extend(["--port", str(port)])
+
+    if "--model" not in arg_names:
+        vllm_cmd.extend(["--model", os.fspath(model_path)])
+
+    # Force multiprocessing for distributed serving, vLLM will try "ray" if it's installed but we do
+    # not support it (yet?). We don't install Ray but we might end up running on systems that have it,
+    # so let's make sure we use multiprocessing.
+    if "--distributed-executor-backend" not in arg_names:
+        vllm_cmd.extend(["--distributed-executor-backend", "mp"])
```

Comment:
`mp` is actually the default since vLLM version 0.5.0, even if ray is installed.

It will only use ray if already running within a ray cluster, or for multi-node scenarios where mp isn't yet supported.

There's no harm in keeping this though just in case an earlier vLLM version is used...

Reply:
@n1hility I remember you mentioning the project might consider switching to upstream v0.5.1. Do you think we should make a note that this should be removed once the version of vLLM is bumped past that?

Reply:
While doing this, I didn't look at the code to confirm but this is what the doc says:

> Backend to use for distributed serving. When more than 1 GPU is used, will be automatically set to “ray” if installed or “mp” (multiprocessing) otherwise.

That's the reason behind patch in the first place. Ultimately, I guess it doesn't hurt.

Reply:
I think the docs on this are wrong as for me it no longer picks ray even when installed. I agree though it's nice to be sure.